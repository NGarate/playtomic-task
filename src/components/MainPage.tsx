import React from 'react';
import { BrowserRouter as Router, Switch, Redirect, Route, Link } from "react-router-dom";
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Dashboard from './Dashboard';
import Settings from './Settings';

const drawerWidth = 240;

const useStyles = makeStyles(({mixins, palette, spacing}: Theme) =>
  createStyles({
    root: {
      display: 'flex',
    },
    appBar: {
      width: `calc(100% - ${drawerWidth}px)`,
      marginLeft: drawerWidth,
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0,
    },
    drawerPaper: {
      width: drawerWidth,
    },
    toolbar: mixins.toolbar,
    content: {
      flexGrow: 1,
      backgroundColor: palette.background.default,
      padding: spacing(3),
      marginLeft: drawerWidth
    },
  }),
);


export default function MainPage() {
    const classes = useStyles();

    return (
        <Router>
            <AppBar position="fixed" className={classes.appBar}>
                <Toolbar>
                    <Typography variant="h6" noWrap>Playtomic task</Typography>
                </Toolbar>
            </AppBar>
            <Drawer
                className={classes.drawer}
                variant="permanent"
                anchor="left"
                classes={{
                    paper: classes.drawerPaper,
                }}
            >
                <List>
                    <ListItem key="dashboard">
                        <Link to="/">Dashboard</Link>
                    </ListItem>
                    <ListItem key="settings">
                        <Link to="/settings">Settings</Link>
                    </ListItem>
                </List>
            </Drawer>
            <main className={classes.content}>
                <div className={classes.toolbar} />
                <Switch>
                    <Route exact path="/" component={Dashboard} />
                    <Route path="/settings" component={Settings} />
                    <Redirect to="/" />
                </Switch>
            </main>
        </Router>
    );
}
