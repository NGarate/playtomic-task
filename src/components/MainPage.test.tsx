import React from 'react';
import toJson from 'enzyme-to-json';
import MainPage from './MainPage';
import * as redux from 'react-redux';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

jest.mock('react-redux');
configure({ adapter: new Adapter() });

test('Main page renders correctly when user is authenticated', () => {
    const mockedDispatch = () => {
        // empty function
    };
    redux.useSelector.mockReturnValue({
        isAuthenticated: true,
        user: { userName: 'userName' }
    });
    redux.useDispatch.mockImplementation(() => mockedDispatch);

    const wrapper = shallow(<MainPage />);

    expect(toJson(wrapper)).toMatchSnapshot();
});

test('Main page renders correctly when user is not authenticated', () => {
    const mockedDispatch = () => {
        // empty function
    };
    redux.useSelector.mockReturnValue({
        isAuthenticated: false,
        user: { userName: 'userName' }
    });
    redux.useDispatch.mockImplementation(() => mockedDispatch);

    const wrapper = shallow(<MainPage />);

    expect(toJson(wrapper)).toMatchSnapshot();
});
