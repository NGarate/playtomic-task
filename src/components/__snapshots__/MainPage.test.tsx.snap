// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page renders correctly when user is authenticated 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      path="/login"
    />
    <ProtectedRoute
      component={[Function]}
      exact={true}
      path="/"
    />
    <ProtectedRoute
      component={[Function]}
      path="/settings"
    />
    <Redirect
      to="/login"
    />
  </Switch>
</BrowserRouter>
`;

exports[`Main page renders correctly when user is not authenticated 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      path="/login"
    />
    <ProtectedRoute
      component={[Function]}
      exact={true}
      path="/"
    />
    <ProtectedRoute
      component={[Function]}
      path="/settings"
    />
    <Redirect
      to="/login"
    />
  </Switch>
</BrowserRouter>
`;
